src/goto-harness/memory_snapshot_harness_generator.cpp

```diff
@@ -232,57 +276,19 @@ void memory_snapshot_harness_generatort::get_memory_snapshot(
   symbol_table_from_json(json, snapshot);
 }
 
-void memory_snapshot_harness_generatort::generate(
-  goto_modelt &goto_model,
-  const irep_idt &harness_function_name)
+void memory_snapshot_harness_generatort::generate(goto_modelt &goto_model)
 {
   symbol_tablet snapshot;
-  get_memory_snapshot(memory_snapshot, snapshot);
+  get_memory_snapshot(memory_snapshot_file, snapshot);
 
   symbol_tablet &symbol_table = goto_model.symbol_table;
   goto_functionst &goto_functions = goto_model.goto_functions;
 
-  const symbolt *called_function_symbol = symbol_table.lookup(function);
-
-  if(called_function_symbol == nullptr)
-  {
-    throw invalid_command_line_argument_exceptiont(
-      "function `" + id2string(function) + "` not found in the symbol table",
-      "--initial-location");
-  }
-
-  {
-    const auto f_it = goto_functions.function_map.find(function);
-
-    if(f_it == goto_functions.function_map.end())
-    {
-      throw invalid_command_line_argument_exceptiont(
-        "goto function `" + id2string(function) + "` not found",
-        "--initial-location");
-    }
-
-    if(!f_it->second.body_available())
-    {
-      throw invalid_command_line_argument_exceptiont(
-        "given function `" + id2string(function) + "` does not have a body",
-        "--initial-location");
-    }
-  }
+  const symbolt *called_function_symbol = symbol_table.lookup(entry_function_name);
 
   add_init_section(goto_model);
 
-  if(symbol_table.has_symbol(harness_function_name))
-  {
-    throw invalid_command_line_argument_exceptiont(
-      "harness function `" + id2string(harness_function_name) +
-        "` already in "
-        "the symbol table",
-      "--" GOTO_HARNESS_GENERATOR_HARNESS_FUNCTION_NAME_OPT);
-  }
-
-  code_blockt harness_function_body;
-
-  add_assignments_to_globals(snapshot, harness_function_body);
+  code_blockt harness_function_body = add_assignments_to_globals(snapshot);
```

Comment:
I think this could be better with a comment to explain intent. The name and signature are not immediately obvious what they do.